src/coreclr/src/pal/src/loader/module.cpp

```diff
@@ -963,6 +963,8 @@ BOOL LOADInitializeModules()
     exe_module.prev = &exe_module;
     exe_module.pDllMain = nullptr;
     exe_module.hinstance = nullptr;
+    exe_module.pDllMain = (PDLLMAIN)dlsym(exe_module.dl_handle, "DllMain");
```

Comment:
```suggestion
    exe_module.pDllMain = (PDLLMAIN)dlsym(exe_module.dl_handle, "DllMain");
    if (exe_module.pDllMain == nullptr)
    {
        ERROR("Boom!\n");
        return FALSE;
    }
```

Reply:
That would cause the normal "hosts" (dotnet, corerun, etc.) to fail because they don't have a DllMain.